src/NUnitFramework/tests/Assertions/AssertWarnTests.cs

```diff
@@ -51,5 +52,47 @@ public void AssertWarnWorksWithMessageAndArgs()
             Assert.AreEqual(ResultState.Warning, result.ResultState);
             Assert.AreEqual("MESSAGE: 2+2=4", result.Message);
         }
+
+        [Test]
+        public void WarningsAreDisplayedWithFailure()
+        {
+            ITestResult result = TestBuilder.RunTestCase(
+                typeof(WarningFixture),
+                "TwoWarningsAndFailure");
+
+            Assert.That(result.ResultState, Is.EqualTo(ResultState.Failure));
+            Assert.That(result.AssertionResults.Count, Is.EqualTo(3));
+            Assert.That(result.Message, Contains.Substring("First warning"));
+            Assert.That(result.Message, Contains.Substring("Second warning"));
+            Assert.That(result.Message, Contains.Substring("This fails"));
+        }
+
+        [Test, Ignore("Currently Fails: Ignored message is displayed without the warnings")]
```

Comment:
@CharliePoole Could you start an issue and gather the information into one place so we can understand how to get to work on these new TODO items?

Reply:
@jnm2 I take it that by TODOs you mean the two ignored tests in this class? Do we know that the tests still don't work?

Reply:
They both still fail on master if I unignore them.

Reply:
Yes. They both still fail on master if I unignore them.

Reply:
@CharliePoole Three TODOs were introduced by this PR. https://github.com/nunit/nunit/pull/2924 handles one which is not related to these two, as far as I can see. I'd appreciate having the information about how to follow up in a new assignable issue, if you're willing and any of this is still fresh in your mind.

Reply:
I'm totally confused now! AFAIK a TODO is a formatted comment that starts `// TODO`, which is specially handled by Visual Studio. I also understand that "todo" is a generic phrase for something that needs to be done. Can you tell me what you mean by it? Where are the three todos? Can I read them?

Earlier in the thread, you asked me to create an issue, which is a term we do have a common understanding of (I think), but now you seem to have resolved the issues in a PR.

I'm almost 100% sure that you are using a term you think I'll understand 😈 but it's not working!

Reply:
I'm so sorry! Replace "TODO" with "Ignore" and the above should make sense.
I had internalized that "Ignore" means "use sparsely and over short periods of time," but I hadn't internalized that this was different than a TODO.